### PR TITLE
system variables: add charset documentation

### PR DESCRIPTION
**Pull request by en-jin19**
### What is changed, added or deleted? (Required)
added charset documentation in system-variables document

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: [#5867](https://github.com/pingcap/docs/pull/5867)
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by en-jin19** [comment]
I have finished self-check, please take a look, thanks! @TomShawn

File: system-variables.md [hunk 1]
```diff
@@ -82,6 +82,37 @@ mysql> SELECT * FROM t1;
...
+
+- 作用域: SESSION | GLOBAL
+- 默认值: `utf8mb4`
+- 这个变量表示客户端的字符集。如果要查看更多 TiDB 中支持的字符集和排序规则，请参见[字符集和排序规则](/character-set-and-collation.md)。 如果需要更改字符集，建议使用[`SET NAMES`](/sql-statements/sql-statement-set-names.md).
+
+### character_set_connection
+
+- 作用域: SESSION | GLOBAL
+- 默认值: `utf8mb4`
+- 这个变量表示没有特定字符集的字符串常量。
```
Comment (TomShawn) [inline system-variables.md:95]:
```suggestion
- 若没有为字符串常量指定字符集，该变量表示这些字符串常量所使用的的字符集。
```

File: system-variables.md [hunk 2]
```diff
@@ -82,6 +82,37 @@ mysql> SELECT * FROM t1;
...
+
+### character_set_database
+
+- 作用域: SESSION | GLOBAL
+- 默认值: `utf8mb4`
+- 这个变量表示当前使用中的默认数据库的字符集，**不建议设置这个变量**。当选择一个新的默认数据库时，服务器会更改该变量的值。
```
Comment (TomShawn) [inline system-variables.md:101]:
```suggestion
- 该变量表示当前默认在用数据库的字符集，**不建议设置该变量**。选择新的默认数据库后，服务器会更改该变量的值。
```

File: system-variables.md [hunk 3]
```diff
@@ -82,6 +82,37 @@ mysql> SELECT * FROM t1;
...
+
+### character_set_results
+
+- 作用域: SESSION | GLOBAL
+- 默认值: `utf8mb4`
+- The character set that is used when data is sent to the client.
```
Comment (TomShawn) [inline system-variables.md:107]:
```suggestion
```

File: system-variables.md [hunk 4]
```diff
@@ -82,6 +82,37 @@ mysql> SELECT * FROM t1;
...
+- 当数据发送至客户端后，该变量则会被使用。
```
Comment (TomShawn) [inline system-variables.md:108]:
```suggestion
- 该变量表示数据发送至客户端时所使用的字符集。
```

File: system-variables.md [hunk 5]
```diff
@@ -82,6 +82,37 @@ mysql> SELECT * FROM t1;
...
+
+### character_set_server
+
+- 作用域: SESSION | GLOBAL
+- 默认值: `utf8mb4`
+- 当`CREATE SCHEMA`中没有设置特定的字符集时，该变量用于创建新的`SCHEMA` 。
```
Comment (TomShawn) [inline system-variables.md:114]:
```suggestion
- 当 `CREATE SCHEMA` 中没有指定字符集时，该变量表示这些新建的表结构所使用的字符集。
```

File: system-variables.md (hunk 3 above)
Comment (TomShawn) [reply system-variables.md:107]: @en-jin19 中文文档里不应该出现英文原句，请删掉~

**Review by wjhuang2016 — APPROVED** [review]

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**